va/validation-authority.go

```diff
@@ -447,24 +453,30 @@ func (va *ValidationAuthorityImpl) getCAASet(hostname string) (*CAASet, error) {
 			break
 		}
 
-		// Query CAA records for domain and its alias if it has a CNAME
-		for _, alias := range []bool{false, true} {
-			if alias {
-				target, _, err := va.DNSResolver.LookupCNAME(queryDomain)
-				if err != nil {
-					return nil, err
-				}
-				queryDomain = target
-			}
-			CAAs, _, err := va.DNSResolver.LookupCAA(queryDomain)
+		cnames := 0
+		for {
```

Comment:
With this construction the CNAME query will always be made first, altering `querydomain` with the content of the target, in order to abide by the CAA RFC we should ONLY be making CNAME queries for `target` after we have received an empty CAA response for that target. ([Section 4 RFC6844](https://tools.ietf.org/html/rfc6844#section-4)).